python/dgl/heterograph.py

```diff
@@ -6065,6 +6065,88 @@ def local_scope(self):
         self._node_frames = old_nframes
         self._edge_frames = old_eframes
 
+    def new_formats(self, allowed_formats=None, create_formats=None):
```

Comment:
Another question is if it suffices to return a graph with a randomly selected allowed format rather than require users to specify `create_formats`.